tests/testthat/demos/coverage.md

```diff
@@ -0,0 +1,104 @@
+    #> Joining with `by = join_by(id, company_name)`
+    to_edit
+    #> # A tibble: 18 × 15
+    #>       id compa…¹ postc…² country misc_…³ compa…⁴ id_tilt compa…⁵ misc_…⁶ compa…⁷
+    #>    <dbl> <chr>   <chr>   <chr>   <chr>   <chr>     <dbl> <chr>   <chr>   <chr>  
+    #>  1     1 Peasan… 01234   germany A       peasan…       1 Peasan… A       peasan…
+    #>  2     1 Peasan… 01234   germany A       peasan…       2 Peasan… Z       peasan…
+    #>  3     1 Peasan… 01234   germany A       peasan…       4 Peasan… B       peasan…
+    #>  4     2 Peasan… 01234   germany Z       peasan…       1 Peasan… A       peasan…
+    #>  5     2 Peasan… 01234   germany Z       peasan…       2 Peasan… Z       peasan…
+    #>  6     2 Peasan… 01234   germany Z       peasan…       4 Peasan… B       peasan…
+    #>  7     3 Peasan… 11234   germany Z       peasan…       3 Peasan… Z       peasan…
+    #>  8     4 Peasan… 01234   germany Z       peasan…       4 Peasan… B       peasan…
+    #>  9     4 Peasan… 01234   germany Z       peasan…       1 Peasan… A       peasan…
+    #> 10     4 Peasan… 01234   germany Z       peasan…       2 Peasan… Z       peasan…
+    #> 11     5 Bread … 23456   germany C       breadb…       5 The Br… C       thebre…
+    #> 12     6 Flower… 34567   germany Z       flower…       7 Flower… F       flower…
+    #> 13     6 Flower… 34567   germany Z       flower…       6 Flower… D       flower…
+    #> 14     7 Screwd… 45678   germany D       screwd…      NA <NA>    <NA>    <NA>   
+    #> 15     8 Screwd… 45678   germany Z       screwd…      NA <NA>    <NA>    <NA>   
+    #> 16     9 John M… 56789   germany E       johnme…       8 John a… E       johnja…
+    #> 17    10 John M… 55555   germany Y       johnme…      NA <NA>    <NA>    <NA>   
+    #> 18    11 John M… 55555   norway  Y       johnme…      NA <NA>    <NA>    <NA>   
+    #> # … with 5 more variables: postcode_tilt <chr>, country_tilt <chr>,
+    #> #   similarity <dbl>, suggest_match <lgl>, accept_match <lgl>, and abbreviated
+    #> #   variable names ¹​company_name, ²​postcode, ³​misc_info, ⁴​company_alias,
+    #> #   ⁵​company_name_tilt, ⁶​misc_info_tilt, ⁷​company_alias_tilt
+
+## Share of matched companies after manual validation from total loanbook companies
+
+    accepted <- to_edit %>%
+      filter(accept_match)
+    accepted
+    #> # A tibble: 0 × 15
+    #> # … with 15 variables: id <dbl>, company_name <chr>, postcode <chr>,
+    #> #   country <chr>, misc_info <chr>, company_alias <chr>, id_tilt <dbl>,
+    #> #   company_name_tilt <chr>, misc_info_tilt <chr>, company_alias_tilt <chr>,
+    #> #   postcode_tilt <chr>, country_tilt <chr>, similarity <dbl>,
+    #> #   suggest_match <lgl>, accept_match <lgl>
+
+    matched_rows <- accepted %>% nrow()
+    matched_rows
+    #> [1] 0
+
+    total_rows <- loanbook %>%
+      distinct(id) %>%
+      nrow()
+
+    matched_share <- matched_rows / total_rows
+    matched_share
+    #> [1] 0
```

Comment:
@kalashsinghal, 

We're now using the `accept_match` columns but we're still using the wrong dataset. The dataset `to_edit` has our suggestions but has not yet been validated by the user. In the dataset `to_edit` all values of `accept_match` are `NA` -- which explain why `to_edit %>% filter(accept_match)` returns a 0-row dataframe. I re-rendered the code and printed the objects for you to see the output. 

Coverage needs to be calculated after the user validates the matches and pupulates `accept_match` with `TRUE` -- i.e. at [step 3 in Get started](https://2degreesinvesting.github.io/tilt.company.match/articles/tilt-company-match.html#in-r-session-2). Notice how at step 3 the dataset we use is "demo_matched".

This is an excellent example why it's important to write automatic tests. After some time it's very easy to change the code and not notice that the change brakes something. Ideally we should wrap the code you wrote into some kind of `calculate_coverage()` function for which we can write automated tests.

Reply:
@maurolepore I will create the function `calculate_coverage()` to use automatic testing asap